pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -124,14 +125,14 @@ func (o createAuthInfoOptions) run() error {
 }
 
 // authInfo builds an AuthInfo object from the options
-func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
+func (o *createAuthInfoOptions) modifyAuthInfo(cmd *cobra.Command, existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
-		certPath := o.clientCertificate.Value()
-		if o.embedCertData.Value() {
+	if cmd.Flags().Changed(clientcmd.FlagCertFile) {
```

Comment:
I actually don't mind the primitives in the struct in general. In retrospect, an explicit "unset" or "clear" option would have removed the need to call Provided or Changed, etc. But since part of the input to the command is whether these fields were provided, that input should be captured in the command options struct.